File: KataWordWrap.MSTest/Wrapper.cs
namespace KataWordWrap.MSTest
{
  public static class Wrapper {
    public static string Wrap(string text, int marker) {
      const string newline = "\n";

      if (text.Length <= marker) return text;

      string line = text.Substring(0, marker + 1);
      int lastSpaceOnLine = line.LastIndexOf(" ");

      if(lastSpaceOnLine > 0)
        return text.Substring(0, lastSpaceOnLine) + newline + Wrap(text.Substring(lastSpaceOnLine + 1), marker);

      return text.Substring(0, marker) + newline + Wrap(text.Substring(marker), marker);
    }
  }
}

File: KataWordWrap.Jasmine/public/Wrapper.js
var wrap = function(text, marker){
  if (text.length < marker)
    return text;

  var lastspace = text.lastIndexOf(' ', marker);
  var wrapat = lastspace > 0 ? lastspace : marker;
  
  return text.substring(0, wrapat).trim() + '\n' + text.substring(wrapat).trim();
};
